src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs

```diff
@@ -17,6 +17,370 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+
+namespace System.Runtime.InteropServices
+{
+    [AttributeUsage(AttributeTargets.Parameter, Inherited = false)]
+    public sealed class OptionalAttribute : Attribute
+    {
+        public OptionalAttribute()
+        {
+        }
+    }
+
+    [AttributeUsage(AttributeTargets.Parameter)]
+    public sealed class DefaultParameterValueAttribute : Attribute
+    {
+        public DefaultParameterValueAttribute(object value)
+        {
+            Value = value;
+        }
+        public object Value { get; }
+    }
+}
+
+class Program
+{
+    const string s2 = nameof(s2);
+    delegate void D([CallerArgumentExpression(s2)] [Optional] [DefaultParameterValue(""default"")] ref string s1, string s2);
+
+    static void M(ref string s1, string s2)
+    {
+    }
+
+    public static void Main()
+    {
+        D d = M;
+        string s = string.Empty;
+        d.EndInvoke(ref s, null);
+    }
+}
+";
+
+            var compilation = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.ReleaseExe, parseOptions: TestOptions.RegularPreview);
+            compilation.VerifyDiagnostics(
+                // (29,22): error CS9006: The CallerArgumentExpressionAttribute may only be applied to parameters with default values
+                //     delegate void D([CallerArgumentExpression(s2)] [Optional] [DefaultParameterValue("default")] ref string s1, string s2);
+                Diagnostic(ErrorCode.ERR_BadCallerArgumentExpressionParamWithoutDefaultValue, "CallerArgumentExpression").WithLocation(29, 22));
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void TestEndInvoke3()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices
+{
+    [AttributeUsage(AttributeTargets.Parameter, Inherited = false)]
+    public sealed class OptionalAttribute : Attribute
+    {
+        public OptionalAttribute()
+        {
+        }
+    }
+
+    [AttributeUsage(AttributeTargets.Parameter)]
+    public sealed class DefaultParameterValueAttribute : Attribute
+    {
+        public DefaultParameterValueAttribute(object value)
+        {
+            Value = value;
+        }
+        public object Value { get; }
+    }
+}
+
+class Program
+{
+    const string s1 = nameof(s1);
+    delegate void D(string s1, [CallerArgumentExpression(s1)] ref string s2 = ""default"");
+
+    static void M(string s1, ref string s2)
+    {
+    }
+
+    public static void Main()
+    {
+        D d = M;
+        string s = string.Empty;
+        d.EndInvoke(ref s, null);
+    }
+}
+";
+
+            var compilation = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.ReleaseExe, parseOptions: TestOptions.RegularPreview);
+            compilation.VerifyDiagnostics(
+                // (28,33): error CS9006: The CallerArgumentExpressionAttribute may only be applied to parameters with default values
```

Comment:
>// (28,33): error CS9006: The CallerArgumentExpressionAttribute may only be applied to parameters with default values [](http://example.com/codeflow?start=16&length=117)

Do we want to suppress this error, since we already reported (going to report) ERR_RefOutDefaultValue?

Reply:
I'm not sure it's worth changing the behavior, especially that this behavior already exists with CallerMemberName and other caller infos. But I'll change if you think it's worth it :)

Reply:
> I'm not sure it's worth changing the behavior, especially that this behavior already exists with CallerMemberName and other caller infos. But I'll change if you think it's worth it

I do not have strong opinion on that. Given that twe have "preexisting condition", I can live with the current behavior.